ui/src/components/Terminal/terminal.tsx

```diff
@@ -0,0 +1,96 @@
+import { Terminal } from '@xterm/xterm'
+import { Button, Modal } from 'antd'
+import React from 'react'
+
+export interface ITerminal {
+  terminalId: string
+  onClose?: () => void
+  onConnected?: () => void
+}
+
+function devLog(...args: any[]) {
+  if (process.env.NODE_ENV === 'development') {
+    console.log(...args)
+  }
+}
+
+export const OBTerminal: React.FC<ITerminal> = (props) => {
+  const { terminalId } = props
+  const ref = React.useRef<HTMLDivElement>(null)
+  const [ws, setWs] = React.useState<WebSocket | null>(null)
+
+  React.useEffect(() => {
+    if (ref.current) {
+      const term = new Terminal({
+        cols: 160,
+        rows: 60,
+      })
+      term.open(ref.current)
+
+      if (term.options.fontSize) {
+        const containerWidth = ref.current.clientWidth
+
+        const cols = Math.floor(containerWidth / 9.2)
```

Comment:
How are 9 points 2 and 4 calculated?

Reply:
I got the magic ratio through some experiments. `9.2` gave a good effect.